capstone/src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.sps.servlets;
+import java.io.IOException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Key;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Arrays;
+import com.google.gson.Gson;
+import com.google.sps.data.User;
+import com.google.sps.data.Room;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/search")
+public class SearchServlet extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Key userKey = ((User) request.getAttribute(User.USER_ENTITY_NAME)).getUserKey();
+        String searchRequest = getParameter(request, "search", "");
+        Filter searchFilter = null;
+
+        if (searchRequest != "") {
```

Comment:
This is relying on the paticulars of how you set the default above, but you could instead just call request.getParameter, and compare it to null yourself, since you dont actually want a default, you just want to know if the search parameter wasnt specified.

Reply:
This makes sense. I'm not 100% sure why I used getParameter at the time...